generator/src/test/java/com/scottlogic/deg/generator/inputs/profileviolation/ProfileViolationTests.java

```diff
@@ -45,6 +84,139 @@ public void setUp() throws IOException {
         target = new IndividualRuleProfileViolator(mockManifestWriter, mockPath, ruleViolator);
     }
 
+
+    @ParameterizedTest
+    @MethodSource("allAtomicConstraints")
+    public void violate_withLinearProfileSingleRuleSingleConstraint_producesViolatedProfile(
+        Class<? extends AtomicConstraint> atomicConstraint, Object value
+    ) throws IOException {
+        //Arrange
+        Field fooField = new Field("foo");
+
+        ConstraintChainBuilder<Rule> inputRuleBuilder = new RuleBuilder("Input Rule")
+            .withAtomicConstraint(fooField, atomicConstraint, value);
+
+        Profile inputProfile = new Profile(
+            Collections.singletonList(fooField),
+            Collections.singletonList(inputRuleBuilder.build()),
+            "Input Profile");
+
+        //Act
+        List<Profile> violatedProfiles = target.violate(inputProfile);
+
+        //Assert
+        Profile violatedProfile1 = new ViolatedProfile(
+            inputRuleBuilder.build(),
+            new ProfileFields(Collections.singletonList(fooField)),
+            Arrays.asList(inputRuleBuilder.negate().violate().build()),
+            "Input Profile -- Violating: Input Rule"
+        );
+
+        List<Profile> expectedViolatedProfiles = Collections.singletonList(violatedProfile1);
+
+        assertProfileListsAreEquivalent(violatedProfiles, expectedViolatedProfiles);
```

Comment:
there should probably be a method added to typeEqualityHelper that takes in a single profile

Reply:
There is one, I haven't used it as the violate method will always produce a list of profiles